crates/driver/src/infra/blockchain/gas.rs

```diff
@@ -0,0 +1,75 @@
+/// Wrapper around the gas estimation library.
+/// Also llows to add additional tip to the gas price. This is used to
+/// increase the chance of a transaction being included in a block, in case
+/// private submission networks are used.
+use {
+    super::Error,
+    crate::{domain::eth, infra::mempool},
+    ethcontract::dyns::DynWeb3,
+    gas_estimation::{nativegasestimator::NativeGasEstimator, GasPriceEstimating},
+};
+
+type MaxAdditionalTip = f64;
```

Comment:
If this is not a percentage shouldn't this be an integer value?

Reply:
It's `f64` just for the convenience of usage and consistency. It's easier to work with `f64` inside `GasPriceEstimator`.